.github/workflows/docker.yaml

```diff
@@ -314,17 +314,24 @@ jobs:
       - name: Get canonical image name
         id: name
         run: echo "canonical=$(echo ${{ needs.image.outputs.images }} | sed 's/ .*//')" >> "$GITHUB_OUTPUT"
-      - name: Pull images
-        run: |
-          for ARCH in ${{ inputs.archs }} ; do
-            docker pull "${{ steps.name.outputs.canonical }}-${ARCH}"
-          done
       - name: Create and push manifest
         run: |
-          for ARCH in ${{ inputs.archs }} ; do
-            echo "${{ steps.name.outputs.canonical }}-${ARCH}"
-          done | \
-          xargs docker manifest create "${{ steps.name.outputs.canonical }}"
+          # Sometimes GCR gives random errors. Let's try a few times before giving up.
+          for RETRY in $(seq 10) ; do
+            for ARCH in ${{ inputs.archs }} ; do
+              if ! docker pull "${{ steps.name.outputs.canonical }}-${ARCH}" ; then
+                echo "Attempt #${RETRY} failed."
```

Comment:
Would it be useful to add to the output here that it failed on the pull? Or will that be obvious from other output? Similar question for the lower `echo`

Reply:
Good call. I added `set -x` to this step.